tests/Service/FunctionsTest.php

```diff
@@ -0,0 +1,113 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service;
+
+use Lmc\ApiFilter\AbstractTestCase;
+use Lmc\ApiFilter\Entity\Value;
+use Lmc\ApiFilter\Exception\InvalidArgumentException;
+use Lmc\ApiFilter\Filter\FunctionParameter;
+
+/**
+ * @group unit
+ */
+class FunctionsTest extends AbstractTestCase
+{
+    /** @var Functions */
+    private $functions;
+
+    protected function setUp(): void
+    {
+        $this->functions = new Functions();
+    }
+
+    /**
+     * @test
+     */
+    public function shouldRegisterFullNameFunction(): void
+    {
+        $sql = 'SELECT * FROM person WHERE firstName = \':firstName\' AND surname = \':surname\'';
+        $firstName = 'Jon';
+        $surname = 'Snow';
+
+        $this->functions->register(
+            'fullName',
+            ['firstName', 'surname'],
+            function ($filterable, FunctionParameter $firstName, FunctionParameter $surname) {
+                $this->assertSame('firstName', $firstName->getColumn());
+                $firstNameValue = $firstName->getValue()->getValue();
+                $this->assertSame('Jon', $firstNameValue);
+
+                $this->assertSame('surname', $surname->getColumn());
```

Comment:
If you put these strings (`Jon`, `Snow`, `surname`, `firstname`) into constants, it can be more readable. Plus DRY

Reply:
Well yes.. but I dont think it will really be more readable that way. There are still more of these constant values in other tests.